fix(tasks/ast_codegen): run `cargo fmt` synchronously

Boshen opened the pull request:
relates #4152

This is my wild guess, because it works in my other script https://github.com/oxc-project/oxc-browserslist/blob/143685deb1dec9ffce3bc3ba4d9abf7d3fe300ad/xtask/src/main.rs#L21

Boshen commented:
* **#4181** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4181?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Boshen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

rzvxa commented on tasks/ast_codegen/src/main.rs:
```diff
@@ -217,9 +217,6 @@ pub struct CliOptions {
     dry_run: bool,
-    /// Don't run cargo fmt at the end
-    #[bpaf(switch)]
-    no_fmt: bool,
```
LGTM, But please keep this option. It wasn't added because of the issue at hand.

Boshen committed:
fix(tasks/ast_codegen): run `cargo fmt` synchronously (#4181)

relates #4152

This is my wild guess, because it works in my other script https://github.com/oxc-project/oxc-browserslist/blob/143685deb1dec9ffce3bc3ba4d9abf7d3fe300ad/xtask/src/main.rs#L21
